app/controllers/api/users.php

```diff
@@ -1080,6 +1080,61 @@ function createUser(string $hash, mixed $hashOptions, string $userId, ?string $e
         $response->dynamic(new Document($prefs), Response::MODEL_PREFERENCES);
     });
 
+App::post('/v1/users/:userId/tokens')
+    ->desc('Create session token')
+    ->groups(['api', 'users'])
+    ->label('event', 'users.[userId].tokens.create')
+    ->label('scope', 'users.write')
+    ->label('audits.event', 'tokens.create')
+    ->label('audits.resource', 'user/{request.userId}')
+    ->label('usage.metric', 'tokens.requests.create')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'users')
+    ->label('sdk.method', 'createSessionToken')
+    ->label('sdk.description', '/docs/references/users/create-session-token-session.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_CREATED)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_TOKEN)
+    ->param('userId', '', new UID(), 'User ID.')
+    ->inject('response')
+    ->inject('dbForProject')
+    ->inject('queueForEvents')
+    ->action(function (string $userId, Response $response, Database $dbForProject, Event $queueForEvents) {
+        $user = $dbForProject->getDocument('users', $userId);
+
+        if ($user->isEmpty()) {
+            throw new Exception(Exception::USER_NOT_FOUND);
+        }
+
+        $secret = Auth::tokenGenerator();
+        $expire = DateTime::formatTz(DateTime::addSeconds(new \DateTime(), Auth::TOKEN_EXPIRATION_CONFIRM));
+
+        $token = new Document([
+            '$id' => ID::unique(),
+            'userId' => $user->getId(),
+            'userInternalId' => $user->getInternalId(),
+            'type' => Auth::TOKEN_TYPE_UNIVERSAL,
+            'secret' => Auth::hash($secret), // One way hash encryption to protect DB leak
+            'expire' => $expire,
```

Comment:
I think the expiration time for the universal token needs to be really short to reduce the risk of an unauthorized user using it.

Reply:
Had a think about this and added a default of 15 minutes, alongside an optional parameter `expire` to set the seconds until expiry